routing_common/transit_serdes.hpp

```diff
@@ -0,0 +1,136 @@
+#pragma once
+
+#include "geometry/point2d.hpp"
+
+#include "coding/point_to_integer.hpp"
+#include "coding/read_write_utils.hpp"
+#include "coding/reader.hpp"
+#include "coding/varint.hpp"
+#include "coding/write_to_sink.hpp"
+
+#include "base/assert.hpp"
+#include "base/exception.hpp"
+#include "base/macros.hpp"
+
+#include <cmath>
+#include <cstdint>
+#include <limits>
+#include <sstream>
+#include <string>
+#include <vector>
+
+namespace routing
+{
+namespace transit
+{
+template <typename Sink>
+class Serializer
+{
+public:
+  Serializer(Sink & sink) : m_sink(sink) {}
+
+  void operator()(uint8_t const d, char const * /* name */ = nullptr) { WriteToSink(m_sink, d); }
+  void operator()(uint16_t const d, char const * /* name */ = nullptr) { WriteToSink(m_sink, d); }
+  void operator()(uint32_t const d, char const * /* name */ = nullptr) { WriteToSink(m_sink, d); }
+  void operator()(uint64_t const d, char const * /* name */ = nullptr) { WriteToSink(m_sink, d); }
+  void operator()(double const d, char const * /* name */ = nullptr)
+  {
+    NOTIMPLEMENTED();
+  }
+
+  void operator()(std::string const & s, char const * /* name */ = nullptr)
+  {
+    rw::Write(m_sink, s);
+  }
+
+  void operator()(m2::PointD const & p, char const * /* name */ = nullptr)
+  {
+    m2::PointU const pointU = PointD2PointU(p, POINT_COORD_BITS);
+    (*this)(pointU.x);
+    (*this)(pointU.y);
+  }
+
+  template <typename R>
+  void operator()(R const & r, char const * /* name */ = nullptr)
+  {
+    r.Visit(*this);
+  }
+
+  template <typename T>
+  void operator()(std::vector<T> const & vs, char const * /* name */ = nullptr)
+  {
+    CHECK_LESS(vs.size(), std::numeric_limits<uint32_t>::max(), ());
```

Comment:
You may even write uint64_t with varints.

Reply:
Could I ask you pls to explain what you mean? Do you propose to keep the vector size in mwm as uint64_t? If yes done. And I removed `CHECK_LESS(vs.size(), std::numeric_limits<uint64_t>::max(), ());`.

Reply:
I propose to write vector size as uint64_t varint.

Reply:
Yes. Done.